RocketChat/Rocket.Chat.ReactNative
# Chore: evaluate `ThemeView`

## Proposed changes

- Removes `withTheme` from `ThemeView`
- Converts `ThemeView` into functional component
## Issue(s)	


## How to test or reproduce

1. Go to Settings > Theme
2. Change themes (either `Automatic` / `Light` / `Dark` for the theme or `Dark` / Black` for the dark level)

It should change the theme properly across the app
## Screenshots

## Types of changes




## Checklist



## Further comments
